src/twisted/words/test/test_jabberclient.py

```diff
@@ -16,6 +16,14 @@
 from twisted.words.protocols.jabber.sasl import SASLInitiatingInitializer
 from twisted.words.xish import utility
 
+try:
+    from twisted.internet import ssl
+except ImportError:
```

Comment:
There are lots of other places which do this conditional import.  Can we piggyback on one of them rather than adding bespoke `ImportError`-handling logic?  My concern here is that this type of exception handling is likely to mask cases where there's an actual import-time bug in `twisted.internet.ssl` and I don't want to have to go looking for 20 instances of this if we ever manage to go in and fix it to only fail on defined import errors (i.e. those where `cryptography` isn't available).

Worst case, I'd rather literally import it from another `test_` module for the time being.

Reply:
I suppose this is due to the unconditional import of the `OpenSSL` module in `twisted.internet.ssl` and `twisted.internet._sslverify`. The former module does have a `supported` attribute, but it seems to (at least now) always be set to True, even though for example `twisted.words.protocols.jabber.xmlstream` checks for that.

If the `OpenSSL` module cannot be imported, the whole import fails, wheres I think my tests probably don't even use it.

Can we not make `twisted.internet.ssl` do this conditional import and have it set `supported` to `False`, and `SSL` to `None`?

Reply:
@glyph I'm not sure if this should be addressed as part of this ticket?